### ISSUE TITLE
[a11y Design Review] IIR, Welcome Pathway - pre-Staging

### ISSUE THREAD

**Issue by coforma-jamie** [department-of-veterans-affairs/va.gov-team#95849]
# CAIA A11y Support

> [!NOTE]
> - **Review:** [Figma mockup](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=2091-57304&node-type=canvas&t=lqiR9v5Aonz6kNP4-0)
> - **CAIA Review Date**: 11/12/2024
> - **Review Purpose**: Staging Prep
> - **Reviewed By**: CAIA A11ys 

CAIA A11ys can review the Figma prototype files for desktop and mobile (if available), in order to provide feedback that may help your team successfully prepare for your upcoming research sessions. 

## Next Steps

> [!TIP]
> **DEFINITION OF DONE**: Refer to the task list in this ticket, for actionable items.

## Description

The team is currently developing a Figma prototype, that will be referenced and used when creating the coded prototype for user research. 

### Timeframe
Week of 11/11. 

### No Accessibility Specialist on Team
CAIA accessibility specialists are available to provide support to IIR through this process with accessibility support, as this team does not have an accessibility specialist. 

### Provided via CAIA Intake

- [Product outline](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/veteran-onboarding/product-outline.md)
- [User flows](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1715019181472/206d1983971576ee7696c1984666185a2fbff545?fromVisitorModal=true&sender=u4356f0478274fc3cca5b5144)
- [Prototype or mockups](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Welcome-to-My-VA?node-id=2091-57304&node-type=canvas&t=lqiR9v5Aonz6kNP4-0)

## CAIA Assignee(s)

<details><summary>Toggle to view assignee info</summary>
- `tbd`

</details>

## Team Info

<details><summary>Toggle to view team info</summary>

- Team name: IIR (Iterate, Innovate, Run)
- OCTO-DE product owner: Jeff Barns
- Product manager: Pete Egan
- Designer: Liza McRuer/Raquel Eisele
- FE engineer: John Rodriguez/Nick Sayre
- Product/team Slack channel: #iir-product-teams-public

</details>

## Definition of Done

```[tasklist]
### A11y Tasks
- [x] **CAIA Intake Opened** `9/13/2024`
- [x] Receive Figma Prototype from Raquel the week of 9/30/2024
- [ ] Create a Figma Prototype Review Google Doc `very few changes from last time, so only reviewed the changes and didn't need a doc`
- [x] Review of the File
- [ ] **Complete** the review and share findings
- [x] **Additional Support** [Review the menu of services with the team](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/teams/CAIA/accessibility#services-we-offer).
```

```[tasklist]
## Ticket Updates (CAIA Internal)
- [x] _Connect to an `Epic` or `Intake` (what body of work is this a part of?)_
- [x] _Label with `Originator/Team` (product team or stakeholder requesting support)_
- [x] _Label date in the `Open Date` field_
- [x] _Label with `Estimate` (level of effort expected for this ticket)_
- [x] _Add `Assignee(s)` name(s) to ticket_
- [x] _Add `Assignee(s) name(s) to each task` they will complete via handle tag (if known)_
- [x] _Select a `Priority Level`_
- [x] _Update date in `Last Checked` field_
- [ ] _Label with `Actual` (level of effort it took to complete this ticket)_
- [ ] _Update date in_ `Closed  Date`
```

**Comment by coforma-jamie** [comment]
Reviewing your changes after our initial review, I have a few pieces of feedback to share:

- Consider moving the additional info component **below** the content information block, since it contains supplementary information - the "Edit" buttons are more important.
![Image](https://github.com/user-attachments/assets/4a6e0458-0542-440e-bb67-76a76c9a0794)
- Whenever a user selects "Save," they're taken to a new screen, and a success slim alert appears letting them know that they've successfully updated their email address etc., make sure that focus is managed to go to the alert on load.
![Image](https://github.com/user-attachments/assets/1c6062a7-841e-4372-b16d-577f11cdfa49)
- If possible, trigger error messages on "Save," **not** on text entry. It can be really jarring for people to get an error before they've confirmed their changes, and for screen reader users, the error will interrupt their typing and make it hard to complete.
![Image](https://github.com/user-attachments/assets/f733a8e6-4f2f-4f5d-bfc3-b2573cf93b09)
- In situations like these, the "Question" is an H2, and "Mobile phone number..." is the `<label>` for the text input.
![Image](https://github.com/user-attachments/assets/a4d866ca-8a1a-4f68-8827-141b893e1e45)

And a UX question:
- If a user lives abroad (say, on a base), and has a non-US mobile phone number, what should they do in this situation? Should there be an additional info component below, "What to do if you don't have a US mobile phone number?" Thinking....
![Image](https://github.com/user-attachments/assets/c05376f9-7448-4d1a-8716-18fadeabd64a)

cc @raquelou

**Comment by coforma-jamie** [comment]
@sara-amanda ready for closure review

**Comment by raquelou** [comment]
Our team has reviewed the feedback and we have determined, after some discussion with Jamie, that the lack of direction for users with an international phone number and the error messages that appear as a user types can be addressed at a later time. This is due to the fact that our team did not build this functionality but are using a pre-existing form flow.

We believe these changes should not just be made to our singular use case but to the forms flow as a whole.
